src/vss_client.rs

```diff
@@ -0,0 +1,83 @@
+use std::error::Error;
+
+use ::prost::Message;
+use reqwest;
+use reqwest::Client;
+
+use crate::vss::{
+	GetObjectRequest, GetObjectResponse, ListKeyVersionsRequest, ListKeyVersionsResponse, PutObjectRequest,
+	PutObjectResponse,
+};
+use crate::vss_error::VssError;
+
+/// Thin-client to access a hosted instance of Versioned Storage Service (VSS).
+/// The provided [`VssClient`] API is minimalistic and is congruent to the VSS server-side API.
+pub struct VssClient {
+	base_url: String,
+	client: Client,
+}
+
+impl VssClient {
+	/// Constructs a [`VssClient`] using `base_url` as the VSS server endpoint.
+	pub fn new(base_url: &str) -> Result<Self, Box<dyn Error>> {
+		let client = Client::new();
+		Ok(Self { base_url: String::from(base_url), client })
+	}
+
+	/// Fetches a value against a given `key` in `request`.
+	/// Makes a service call to the `GetObject` endpoint of the VSS server.
+	/// For API contract/usage, refer to docs for [`GetObjectRequest`] and [`GetObjectResponse`].
+	pub async fn get_object(&self, request: &GetObjectRequest) -> Result<GetObjectResponse, VssError> {
+		let url = format!("{}/getObject", self.base_url);
+
+		let raw_response = self.client.post(url).body(request.encode_to_vec()).send().await?;
+		let status = raw_response.status();
+		let payload = raw_response.bytes().await?;
+
+		if status.is_success() {
+			let response = GetObjectResponse::decode(&payload[..])?;
+			Ok(response)
+		} else {
+			Err(VssError::new(status, payload))
+		}
```

Comment:
What's the plan for testing?

@tnull I see there is: https://docs.rs/reqwest_mock/latest/reqwest_mock/. Is there a preferred way of going abou this?